### PR TITLE
ROC-4310: Submit claim paid in full

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
[ROC-4310](https://tools.hmcts.net/jira/browse/ROC-4310)

### Change description ###
This PR adds endpoint to store claim paid in full and fires notification to staff and defendant.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/functional/citizen/PaidInFullTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import uk.gov.hmcts.cmc.claimstore.idam.models.User;
+import uk.gov.hmcts.cmc.claimstore.tests.BaseTest;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.PaidInFull;
+
+import java.time.LocalDate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PaidInFullTest extends BaseTest {
```
Comment (singh-surjeet) [inline src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/functional/citizen/PaidInFullTest.java]: How about adding a test case when paid in full is attempted on claim/case 
1. which doesn't exist in database?
2. When user is not a owner of the case?

**Comment by singh-surjeet** [comment]
@sabahirfan please follow link of three sky blue curved line on RHS of failing build which takes you to sonarcloud
screenshot 2018-10-18 at 14 57 41. Your build is failing on low test coverage and can easily be fixed by increasing coverage on class -  src/main/java/uk/gov/hmcts/cmc/claimstore/events/paidinfull/PaidInFullEvent.java.

File: src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/functional/citizen/PaidInFullTest.java (hunk 1 above)
Comment (sabahirfan) [reply src/aat/java/uk/gov/hmcts/cmc/claimstore/tests/functional/citizen/PaidInFullTest.java]: These scenarios are covered in unit tests. Added functional tests as well.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java [hunk 2]
```diff
@@ -308,6 +313,30 @@ public void saveDefendantResponse(
...
+        Claim claim = getClaimByExternalId(externalId, authorisation);
+        String claimantId = userService.getUserDetails(authorisation).getId();
+        assertPaidInFull(claim, claimantId);
+        this.caseRepository.paidInFull(claim, paidInFull, authorisation);
+        Claim updatedClaim = getClaimByExternalId(externalId, authorisation);
+        this.eventProducer.createPaidInFullEvent(updatedClaim);
+        return updatedClaim;
+    }
+
+    public void assertPaidInFull(Claim claim, String claimantId) {
```
[the remark this answers is not in the document]
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java]: no

**Pull request merged**